pkg/aws/aws_client/role.go

```diff
@@ -292,3 +309,58 @@ func completeRolePolicyDocument(statement map[string]interface{}) (string, error
 	assumeRolePolicyDocument, err := json.Marshal(rolePolicyDocument)
 	return string(assumeRolePolicyDocument), err
 }
+
+func (client *AWSClient) AttachPolicy(roleName string, policyArn string, retries int) error {
+	policyAttach := iam.AttachRolePolicyInput{
+		PolicyArn: &policyArn,
+		RoleName:  &roleName,
+	}
+	_, err := client.IamClient.AttachRolePolicy(context.TODO(), &policyAttach)
+	if err != nil {
+		return err
+	}
+
+	start := 0
+	if retries <= 0 {
+		retries = 1
+	}
+
+	for start < retries {
+		attached, err := client.PolicyAttachedToRole(roleName, policyArn)
+		if err != nil && start == retries {
+			return err
+		}
+		if attached {
+			return nil
+		}
+		time.Sleep(1 * time.Minute)
```

Comment:
@OAharoni-RedHat I still feel it's better to give the caller control over this. It makes it much clearer that there is a potential for a significant sleep here. At the simplest, we could add a `retryIntervalInSeconds` parameter to the method.